### ISSUE TITLE
@tinyhttp/markdown - `recursive` option doesn't work

### ISSUE THREAD

**Issue by downrunner-help** [talentlessguy/tinyhttp#155]
**Describe the bug**

I do not think the recursive flag is working. I am trying to serve .md from `docs` and below. Files in the `docs` server fine. Subdirectories are not navigated. I get Not Found errors.

**To Reproduce**

1. Add the below as the sole middleware:

```
app.use(
		md('docs', {
			prefix: '/',
			stripExtension: true,
			recursive: true,
			markedExtensions: [
				{
					headerIds: true,
					gfm: true,
					pedantics: false,
					highlight: function(code, language) {
						const hljs = require('highlight.js'),
							validLanguage = hljs.getLanguage(language) ? language : 'plaintext';
						return hljs.highlight(validLanguage, code).value;
					}
				}
			],
		}))
```

2. Create a directory and file structure

```
docs
   README.md
   news
      subfile.md
```

**Expected behavior**

Getting /README returns transformed Markdown (works)
Getting /news/subfile returns transformed Markdown (fails with a Not Found error)

**Versions**

* `node`: 14
* `@tinyhttp/app`: 0.X

**Additional context**

I have reviewed your source and do not see any obvious errors. However, I am a Vanilla JS person, not a TypeScript person and the TypeScript may be obscuring something.

Thanks for this great software ... now if I could just get Markdown to work ...

**Comment by talentlessguy** [comment]
Hi! Yeah it's most likely my fault, `recursive` prop didn't work in `@tinyhttp/static` either (and that package no longer exists).

Will take a look soon and make a fix

sorry for this inconvenience >_<

and also tbh i need to rewrite `@tinyhttp/markdown`, it doesn't work as it should

**Comment by downrunner-help** [comment]
@talentlessguy thanks for quick response. As a follow-on, I tried using `express-markdown-reloaded`.

```
.use("/", expressmarkdown({
	directory: path.join(__dirname,"/docs"),
	includerawtext: true,
	marked: 
		{
			headerIds: true,
			gfm: true,
			pedantics: false,
			highlight: function(code, language) {
				const validLanguage = hljs.getLanguage(language) ? language : 'plaintext';
				return hljs.highlight(validLanguage, code).value;
			}
		}
}))
```

Not sure who's bug it is, but whatever is passed down the call flow makes tiny crap out:

```
TypeError [ERR_INVALID_ARG_TYPE]: The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received type number (404)
    at Function.from (buffer.js:329:9)
    at createETag (/home/syblackwell/git/downrunner/node_modules/@tinyhttp/send/dist/index.cjs:29:49)
    at ServerResponse.send (/home/syblackwell/git/downrunner/node_modules/@tinyhttp/send/dist/index.cjs:71:43)
    at /home/syblackwell/git/downrunner/node_modules/express-markdown-reloaded/lib/markdown.js:71:28
    at suppressedCallback (fs.js:218:5)
    at FSReqCallback.oncomplete (fs.js:164:23) {
  code: 'ERR_INVALID_ARG_TYPE'
}
```

**Comment by downrunner-help** [comment]
@talentlessguy A bit blunt, but this gets me by for now:

```
function markdownHandler({root="",index="index",markedOptions={}}={}) {
	return async (req,res,next) => {
		const url = req.url==="/" ? index : req.url,
			pathname = path.normalize(path.join(__dirname,root,url.endsWith(".md") ? url : url + ".md"));
		let text;
		try {
			text = await fs.readFile(pathname,"utf8");
			const html = marked(text,markedOptions);
			res.statusCode = 200;
			res.setHeader("content-type","text/html");
			res.end(`<html><body>${html}</body></html>`);
		} catch(e) {
			console.log(e)
			next();
		}
	}
}
```

.get("/*",markdownHandler({
	root: "/docs",
	index: "README",
	markedOptions: {
		headerIds: true,
		gfm: true,
		pedantics: false,
		highlight: function(code, language) {
			const hljs = require("highlight.js"),
				validLanguage = hljs.getLanguage(language) ? language : 'plaintext';
			return hljs.highlight(validLanguage, code).value;
		}
	}
}))

**Comment by talentlessguy** [comment]
@downrunner-help it's fixed now via 756caf9.

Upgrade your `@tinyhttp/markdown` to 0.4.17

```sh
pnpm i @tinyhttp/markdown@0.4.17
```
